### ISSUE TITLE
Cast: Display Video info in Google Home

### ISSUE THREAD

**Issue by londonshliach** [google/shaka-player#2606]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
2.5.12

**Please ask your question**

Usually when casting a video I can go to the google home app and see the name of the video being played as well as a scrub bar to jump backwards / forewards within the video.

Example (from Dailymotion.com):
![image](https://user-images.githubusercontent.com/3040953/83480648-f8264380-a469-11ea-84f1-ea46e41b4cab.png)

But when casting from Shaka player (using the Shaka player Demo page) I only see a volume control and a pause/play button.

Example (from the Shaka Player Demo Page):
![image](https://user-images.githubusercontent.com/3040953/83480666-03796f00-a46a-11ea-9b61-ae4419ad5f7c.png)

I was wondering if there was a way to show the video title and the scrubber as well.

**Comment by theodab** [comment]
At the moment, no. But that's something we could probably add.
I am stuck at home due to quarantine and do not have any ChromeCast devices with me, so I can't test this at the moment. But I'm guessing that would be the `title` field within a `metadata` object, in the [MediaInformation message](https://developers.google.com/cast/docs/reference/messages#MediaInformation) that we use to tell the cast UI about the duration, stream type, etc.
Anyway, we could add some interface to set that on the receiver end. Then you could call that method in the `appDataCallback` or something like that.

**Comment by londonshliach** [comment]
@theodab  Thanks for looking into this.
We are all looking foreward to getting out of the house and back to normal life.

**Comment by joeyparrish** [comment]
I have a Chromecast I can debug with at home.  I'll take a quick look.

**Comment by joeyparrish** [comment]
The lack of scrubber is definitely a bug, since we claim to support that in the source code.  The lack of title information is a little different, though, because it's app-specific.  Our demo should probably support it, but it is up to each app to supply that information.

**Comment by theodab** [comment]
That's why I suggested adding a function to the cast receiver to set that. In the demo, we transmit over the entire asset information inside the `extraData`, which would be enough info to provide a reasonable name. Custom receivers would have to do something similar themselves, of course.

**Comment by joeyparrish** [comment]
Of course.  I'm just pointing out the line between what's a bug and an enhancement, since the symptoms described here are a bit of both.  But since we should have a working scrubber, I'm calling it a bug overall.  I think I will be able to fix it.  I'm testing a few things now.

**Comment by joeyparrish** [comment]
We have fixed the scrubber on the Home app, and that change will be cherry-picked to v2.5.13 if possible.

We have also added APIs to set full content metadata in shaka.cast.CastReceiver.  These will first appear in the v3.0.0 release.

All of these changes are available in the master branch and the [nightly demo](https://nightly-dot-shaka-player-demo.appspot.com/demo/) now.

Below are screenshots of the Home app with these changes.

The fixed scrubber:
![Google Home App with scrubber for seeking](https://user-images.githubusercontent.com/6843156/83676558-bb546c80-a58f-11ea-803b-77b3f6bfe754.png)

The "media" view with richer content metadata:
![Google Home App media view, with title and content thumbnail](https://user-images.githubusercontent.com/6843156/83676624-d7580e00-a58f-11ea-8816-3f591f4b5d13.png)
